sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -975,6 +975,14 @@ private void newTimeout(
         final long idleEndpointTimeoutInNanos,
         final long requestTimerResolutionInNanos) {
 
+        if (idleEndpointTimeoutInNanos == 0) {
```

Comment:
Based on the offline discussion with @xinlian12 , this change is not correct. We are working to rectify it. Blocking the PR until it is fixed and we run the full CI to make sure there is no regression.

Reply:
Thanks for capture it. This method also contains acquisition queue check. Updated